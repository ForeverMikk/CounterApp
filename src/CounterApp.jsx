import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 5 }) => {
  //handleAdd

  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
    // Este se usa en caso de que no se haga una referencia directa
    // setCounter ( (c) => c + 1);
  };

  const handleSubstract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <p>{counter}</p>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubstract}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
