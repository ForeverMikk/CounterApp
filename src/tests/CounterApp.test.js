import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import CounterApp from "../CounterApp";

describe("Pruebas sobre el componente Counter App", () => {
  //En este caso usamos el let para que el wrapper sea una variable global
  let wrapper = shallow(<CounterApp value={5} />);

  //Esta instruccion se va a inicializar cada que se realice una prueba
  beforeEach(() => {
    wrapper = shallow(<CounterApp value={5} />);
  });

  test("debe mostrar Correctamente <CounterApp />", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el valo por defecto como 100", () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);

    const valorPordefecto = wrapper.find("p").text().trim();

    expect(valorPordefecto).toBe("100");
  });

  test("debe de incrementar con el boton de +1", () => {
    // El .at(0) es para indicarle el indice del elemento que estas buscando
    // En este caso se busca el primer boton de 3
    wrapper.find("button").at(0).simulate("click");

    const counterText = wrapper.find("p").text().trim();

    expect(counterText).toBe("6");
  });

  test("debe de reducir el valor con el boton de -1", () => {
    wrapper.find("button").at(2).simulate("click");

    const counterText = wrapper.find("p").text().trim();

    expect(counterText).toBe("4");
  });

  test("debe de regresar el valor por defecto con el boton reset", () => {
    const wrapper = shallow(<CounterApp value={105} />);

    wrapper.find("button").at(2).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");

    const counterText = wrapper.find("p").text().trim();

    expect(counterText).toBe("105");
  });
});
