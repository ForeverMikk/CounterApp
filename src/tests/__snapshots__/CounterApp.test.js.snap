// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente Counter App debe mostrar Correctamente <CounterApp /> 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    5
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
